File: discipline_client_ts/discipline_android_primitive_ts/utilities/textual_error/TextualErrorContext.ts
import { Branded } from "../../x.ts";
import { TextualErrorAttachment } from "./TextualErrorAttachment.ts"

const brand = Symbol();

export type TextualErrorContext = Branded<typeof brand, {
  readonly action: string,
  readonly errorMessages: string[],
  readonly infoMessages: string[],
  readonly attachements: TextualErrorAttachment[],
}>;

export const create = (action: string): TextualErrorContext => {
  return Branded(brand, {
    action,
    attachements: [],
    infoMessages: [],
    errorMessages: [],
  });
};

export const TextualErrorContext = {
  create,
};

File: discipline4/app/src/main/java/com/example/app/TextualErrorContext.kt
package com.example.app

import com.example.app.TextualErrorAttachment;

/** Represents a context for error tracking with action and messages */
public data class TextualErrorContext
private constructor(
  val action: String,
  val errorMessages: MutableList<String> = mutableListOf(),
  val infoMessages: MutableList<String> = mutableListOf(),
  val attachments: MutableList<TextualErrorAttachment> = mutableListOf()
) {
  companion object {
    fun create(action: String): TextualErrorContext {
      return TextualErrorContext(action)
    }
  }

  // Convenience methods for adding messages
  fun addErrorMessage(message: String) {
    errorMessages.add(message)
  }

  fun addInfoMessage(message: String) {
    infoMessages.add(message)
  }

  fun addAttachment(attachment: TextualErrorAttachment) {
    attachments.add(attachment)
  }

  // Check if context has any content
  fun isEmpty(): Boolean {
    return errorMessages.isEmpty() && infoMessages.isEmpty() && attachments.isEmpty()
  }
}
